### PR TITLE
Download validation

### PR DESCRIPTION
**Pull request by Captchah**
TL;DR: This change is meant to help with restarting downloads, without deleting them.
Info:
This started from [a Reddit comment of u/amogus2004](https://www.reddit.com/r/swd_io/comments/uy55qg/comment/iamj5e9/?utm_source=reddit&utm_medium=web2x&context=3), who faced a timeout error on their download. After doing some research, I found that this is a SteamCMD issue, which according to users can be fixed by restarting the download, and "validate" ensures the files do not get deleted (for more information, you can see [my Reddit post](https://www.reddit.com/r/swd_io/comments/uy55qg/comment/ianayw0/)).

I am keeping this as a draft PR because I want to make sure everything works as intended for other people than me. Feel free to ask me to undraft it, though.

Edit: Well, the issue u/amogus2004 was facing was related, in fact, to anonymous downloads (the mod could not be downloaded anonymously). I'm not sure what would happen at the downloading of a large file, and a potential timeout there, and more specifically I'm not sure validating helps, so I'll need to find a huge file, limit my connection and test it.

### CODE REVIEW

**Comment by Captchah** [comment]
So, I was able to restart a download after consecutive timeouts, and the files were successfully downloaded without the need to use the validate parameter, as (at least in my instance) they were not getting deleted after the timeout. In light of this test, I'll be closing the PR.
URL of heavy extension for someone who may want to verify it themselves: https://steamcommunity.com/sharedfiles/filedetails/?id=2301404099

**Comment by shadoxxhd** [comment]
It might be useful to test whether it has any effect on _corrupted_ downloads, but since in that case one could simply delete the downloaded folder and try again, it doesn't seem to be a priority.
